coderd/prebuilds/global_snapshot.go

```diff
@@ -125,20 +125,29 @@ func (s GlobalSnapshot) IsHardLimited(presetID uuid.UUID) bool {
 }
 
 // filterExpiredWorkspaces splits running workspaces into expired and non-expired
-// based on the preset's TTL.
-// If TTL is missing or zero, all workspaces are considered non-expired.
+// based on the preset's TTL and last_invalidated_at timestamp.
+// A prebuild is considered expired if:
+// 1. It was created before the preset's last_invalidated_at timestamp, OR
```

Comment:
nit: I think here we should explain what causes expiration (invalidation or TTL), not how we detect it (timestamp comparison). 
```suggestion
// 1. The preset has been invalidated (last_invalidated_at is set), OR
```